src/libraries/System.Composition.TypedParts/src/System/Composition/TypedParts/Discovery/DiscoveredPart.cs

```diff
@@ -130,6 +135,34 @@ private IEnumerable<CompositionDependency> GetPartActivatorDependencies(Dependen
             }
         }
 
+        private ConstructorInfo GetConstructorInfoFromGenericType(TypeInfo type)
+        {
+            var genericPartType = type.GetGenericTypeDefinition();
+            var genericPartTypeInfo = genericPartType.GetTypeInfo();
+            ConstructorInfo constructor = null;
+
+            int index = -1;
+            foreach (var c in genericPartTypeInfo.DeclaredConstructors)
+            {
+                ++index;
```

Comment:
It's not usual to initialize a counter index with value -1, and then do ++index.

Also, why not do this inline on line 159 where you are using it.

Have you considered perhaps using a `for` rather than `foreach`, if you need a counter like this in your block anyway?